File: Product_of_Array_Except_Self/Product_of_Array_Except_Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int len = nums.size();
        vector<int> ans(len, 1);
        // vector<int> left(len, 1);
        for(int i=1;i<len;i++){
            ans[i] = ans[i-1]*nums[i-1];
        }
        int right = 1;
        for(int i=len-1;i>=0;i--){
            ans[i] *= right;
            right *= nums[i];
        }
        return ans;
    }
};

File: Product_of_Array_Except_Self/Product_of_Array_Except_Self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        length = len(nums)
        left = [1] * length
        for i in range(1, length):
            left[i] = left[i-1]*nums[i-1]
            print(left[i]) 

        right = 1

        for i in range(length-1, -1, -1):
            left[i] = left[i]*right
            right *= nums[i]
        return left
